JOSS submission [do not merge] (#104)
fusion-energy/openmc-plasma-source: paper.md

---
title: 'openmc-plasma-source: Pre-built Fusion Neutron Sources for OpenMC'
tags:
  - Python
  - OpenMC
  - neutron sources
  - fusion
authors:
  - name: Remi Delaporte-Mathurin
    corresponding: true
    orcid: 0000-0003-1064-8882
    affiliation: 1

>   - name: Jon Shimwell

Reviewer: ```suggestion
  - name: Jonathan Shimwell
```

    orcid: 0000-0001-6909-0946
    affiliation: 2

affiliations:
 - name: Plasma Science and Fusion Center, MIT, USA
   index: 1

>  - name: Proxima Fusion, Germanyu

Reviewer: ```suggestion
 - name: Proxima Fusion, Germany
```

   index: 2

date: 26 November 2024
bibliography: paper.bib

---

# Summary

> `openmc-plasma-source` is a Python-based package offering a collection of pre-built neutron sources designed for fusion applications using the Monte Carlo particle transport code OpenMC [@romano_openmc_2015]. By providing ready-to-use implementations for various neutron source configurations, such as tokamak, ring, and point sources, this package simplifies the often complex task of neutron source definition in fusion-related Monte Carlo simulations. These sources are parameterised to account for spatial distributions, plasma temperatures, and fusion fuel compositions, enabling realistic and computationally efficient neutron emission models.

Reviewer: ```suggestion
`openmc-plasma-source` is a Python-based package offering a collection of pre-built neutron sources designed for fusion applications using the Monte Carlo particle transport code OpenMC [@romano_openmc_2015].
By providing ready-to-use implementations for various neutron source configurations, such as tokamak, ring, and point sources, this package simplifies the often complex task of neutron source definition in fusion-related Monte Carlo simulations.
These sources are parameterised to account for spatial distributions, plasma temperatures, and fusion fuel compositions, enabling realistic and computationally efficient neutron emission models.
```

> The package is designed to integrate seamlessly into OpenMC workflows, allowing users to define sources in just a few lines of Python code. It also supports advanced features like temperature-based neutron spectra and spatial source distributions, making it an invaluable tool for researchers simulating neutron behaviour in fusion devices.

Reviewer: ```suggestion
The package is designed to integrate seamlessly into OpenMC workflows, allowing users to define sources in just a few lines of Python code.
It also supports advanced features like temperature-based neutron spectra and spatial source distributions, making it an invaluable tool for researchers simulating neutron behaviour in fusion devices.
```

# Statement of need

> Accurate modelling of neutron sources is critical for fusion energy research, underpinning tasks such as reactor shielding design, material testing, and tritium breeding analysis. In this context, OpenMC is a widely used tool for neutron transport simulations [@romano_openmc_2015]. However, creating realistic neutron source models for fusion applications can be a time-consuming and error-prone process, requiring detailed knowledge of plasma physics and significant coding effort.

Reviewer: ```suggestion
Accurate modelling of neutron sources is critical for fusion energy research, underpinning tasks such as reactor shielding design, material testing, and tritium breeding analysis.
In this context, OpenMC is a widely used tool for neutron transport simulations [@romano_openmc_2015].
However, creating realistic neutron source models for fusion applications can be a time-consuming and error-prone process, requiring detailed knowledge of plasma physics and significant coding effort.
```

> Traditionally, researchers have implemented their own custom neutron source definitions, which often results in redundant work and inconsistencies between studies. For example, spatial distributions, temperature effects, and fuel compositions must be correctly parameterised to ensure the fidelity of the simulations. The lack of standardised tools for these tasks introduces variability and potential errors in simulations.

Reviewer: ```suggestion
Traditionally, researchers have implemented their own custom neutron source definitions, which often results in redundant work and inconsistencies between studies.
For example, spatial distributions, temperature effects, and fuel compositions must be correctly parameterised to ensure the fidelity of the simulations.
The lack of standardised tools for these tasks introduces variability, potential errors in simulations and a lack of reproducibility.
```

`openmc-plasma-source` addresses these challenges by providing a standardised and easy-to-use interface for defining neutron sources in OpenMC. The package implements the equations for neutron distributions based on established models, such as those described in @fausser_tokamak_2012. By automating the setup process and including extensive documentation and examples, it reduces barriers to entry for researchers new to OpenMC or neutron source modelling.

With pre-built configurations for tokamak, ring, and point sources, `openmc-plasma-source` is suitable for a wide range of applications. For example:

- The **tokamak source** models realistic spatial and temperature distributions, optimised for computational efficiency through the use of ring sources.
- The **ring source** offers a simplified yet effective representation for cylindrical geometries.

> - The **point source** is ideal for preliminary studies or cases requiring a concentrated neutron emission.

Reviewer: ```suggestion
- The **point source** is ideal for preliminary studies or cases requiring a concentrated neutron emission found in inertial confinement fusion.
```

---
title: 'openmc-plasma-source: Pre-built Fusion Neutron Sources for OpenMC'
tags:
  - Python
  - OpenMC
  - neutron sources
  - fusion
authors:
  - name: Remi Delaporte-Mathurin
    corresponding: true
    orcid: 0000-0003-1064-8882
    affiliation: 1
  - name: Jonathan Shimwell
    orcid: 0000-0001-6909-0946
    affiliation: 2

affiliations:
 - name: Plasma Science and Fusion Center, MIT, USA
   index: 1
 - name: Proxima Fusion, Germany
   index: 2

date: 26 November 2024
bibliography: paper.bib

---

# Summary

`openmc-plasma-source` is a Python-based package offering a collection of pre-built neutron sources designed for fusion applications using the Monte Carlo particle transport code OpenMC [@romano_openmc_2015].
By providing ready-to-use implementations for various neutron source configurations, such as tokamak, ring, and point sources, this package simplifies the often complex task of neutron source definition in fusion-related Monte Carlo simulations.

> These sources are parameterised to account for spatial distributions, plasma temperatures, and fusion fuel compositions, enabling realistic and computationally efficient neutron emission models.

Reviewer: ```suggestion
These sources are parameterised to account for spatial distributions, plasma temperatures, plasma pressure, and fusion fuel compositions.
DT (deuterium, tritium) plasmas with any ratio of D and T are supported including pure DD and pure TT plasmas.
The package computes spatial distributions of temperature and density and accounts for them when computing the spatially distributed reactivity for the differentfusion reactions availablein the fuel composition.
The temperature, density and fuel composition are accounted for when producing the neutron energy distribution which also varies spatially. 
energy distributions  of neutron sources.
The approach take is also computationally efficient by making use of rotational symmetry to reduce the size of the source definition.
```

tags:
  - Python
  - OpenMC
  - neutron sources
  - fusion
authors:
  - name: Remi Delaporte-Mathurin
    corresponding: true
    orcid: 0000-0003-1064-8882
    affiliation: 1
  - name: Jonathan Shimwell
    orcid: 0000-0001-6909-0946
    affiliation: 2

affiliations:
 - name: Plasma Science and Fusion Center, MIT, USA
   index: 1
 - name: Proxima Fusion, Germany
   index: 2

date: 26 November 2024
bibliography: paper.bib

---

# Summary

`openmc-plasma-source` is a Python-based package offering a collection of pre-built neutron sources designed for fusion applications using the Monte Carlo particle transport code OpenMC [@romano_openmc_2015].
By providing ready-to-use implementations for various neutron source configurations, such as tokamak, ring, and point sources, this package simplifies the often complex task of neutron source definition in fusion-related Monte Carlo simulations.
These sources are parameterised to account for spatial distributions, plasma temperatures, plasma pressure, and fusion fuel compositions.
DT (deuterium, tritium) plasmas with any ratio of D and T are supported including pure DD and pure TT plasmas.

> The package computes spatial distributions of temperature and density and accounts for them when computing the spatially distributed reactivity for the differentfusion reactions availablein the fuel composition.

Reviewer: ```suggestion
The package computes spatial distributions of temperature and density and accounts for them when computing the spatially distributed reactivity for the different fusion reactions possible with the fuel composition.
```

The temperature, density and fuel composition are accounted for when producing the neutron energy distribution which also varies spatially. 

> energy distributions  of neutron sources.

Reviewer: ```suggestion
```

> The approach take is also computationally efficient by making use of rotational symmetry to reduce the size of the source definition.

Reviewer: ```suggestion
The approach taken is also computationally efficient by making use of rotational symmetry to reduce the size of the source definition.
```
